src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -131,13 +131,25 @@ class TaskReceiver(tasksDirectory: String,
             LOGGER.info("Publish $taskId failed with code ${StatusCode.FAILURE}")
         }
 
-        val data = getResultData(taskId, StatusCode.SUCCESS, FileUtils.readFileToByteArray(compiledPdf))
+        val expectedTaskId = getTaskId(request.zipBytes.toByteArray())
+        if (taskId != expectedTaskId) {
```

Comment:
? This code sits inside task compiler, what's the point of checking task id here (besides any sort of "security") ? You want to check task id in the command line client, to make sure that the PDF file which you fetch from Pubsub is the file produced for that very zipped project which was sent by that very launch of client. Imagine 10 clients sending theirs tasks to the fleet, they obviously don't want to steal the results of each other.